---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# SkeletalVis

<!-- badges: start -->
[![R-CMD-check](https://github.com/soulj/SkeletalVis/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/soulj/SkeletalVis/actions/workflows/R-CMD-check.yaml)
<!-- badges: end -->

The goal of SkeletalVis is to enable the search and visualisation of uniformly processed skeletal transcriptomic datasets. It provides tools to identify datasets in which genes of interest are differentially expressed and to discover datasets with similar gene expression profiles to a given query.

It additional allow exploration of the OATargets database and prioritised genes for causing an osteoarthritis damage phenotype in animal models.

## Installation

You can install the development version of SkeletalVis from [GitHub](https://github.com/) with:

``` r
# install.packages("pak")
pak::pak("soulj/SkeletalVis")
```

## Example

This is a basic example which shows how to search for a gene of interest across the SkeletalVis data. See the vignette for a full tutorial of SkeletalVis.

```{r example}
# Load the library
library(SkeletalVis)

# Set up the database
skeletalvis <- load_skeletalvis()

# Get the differential expression of the gene SOX9 across the database
gene_results <- get_gene_fold_changes(skeletalvis, "SOX9", return_fdr = TRUE)
head(gene_results)
```

There are also several built in plotting functions:

```{r plotting}
# Plot the log2fold change against the FDR for SOX9 across the datasets
volcano_plot(gene_results, number_points = 3)
```
